src/Riak/Command/Indexes/Query.php

```diff
@@ -0,0 +1,204 @@
+
+use Basho\Riak\Command;
+use Basho\Riak\CommandInterface;
+
+/**
+ * Class Query
+ *
+ * Riak 2i query information.
+ *
+ * @author Alex Moore <amoore at basho d0t com>
+ */
+class Query extends Command\Object implements CommandInterface
+{
+    /**
+     * @var string
+     */
+    protected $indexName = NULL;
+
+    protected $match = NULL;
+    protected $lowerBound = NULL;
+    protected $upperBound = NULL;
+
+    protected $continuation = NULL; //Binary
+    protected $maxResults; // Int
+    protected $returnTerms; // Bool
+    protected $paginationSort; // Bool
+    protected $termFilter; // string
+    protected $timeout; // timeout
+
+    public function __construct(Command\Builder\QueryIndex $builder)
+    {
+        parent::__construct($builder);
+
+        $this->bucket = $builder->getBucket();
+        $this->indexName = $builder->getIndexName();
+
+        if($builder->isRangeQuery()) {
+            $this->lowerBound = $builder->getLowerBound();
+            $this->upperBound = $builder->getUpperBound();
+        }
+        else {
+            $this->match = $builder->getMatchValue();
+        }
+
+        $this->continuation = $builder->getContinuation();
+        $this->maxResults = $builder->getMaxResults();
+        $this->returnTerms = $builder->getReturnTerms();
+        $this->paginationSort = $builder->getPaginationSort();
+        $this->termFilter = $builder->getTermFilter();
+        $this->timeout = $builder->getTimeout();
+    }
+
+    public function getIndexName() {
+        return $this->indexName;
+    }
+
+    public function getMatchValue() {
+        return $this->match;
+    }
+
+    public function getLowerBound() {
+        return $this->lowerBound;
+    }
+
+    public function getUpperBound() {
+        return $this->upperBound;
+    }
+
+    public function isMatchQuery()
+    {
+        return isset($this->match);
+    }
+
+    public function isRangeQuery()
+    {
+        return isset($this->lowerBound) && isset($this->upperBound);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getContinuation()
+    {
+        return $this->continuation;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getMaxResults()
+    {
+        return $this->maxResults;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function getReturnTerms()
+    {
+        return $this->returnTerms;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function getPaginationSort()
+    {
+        return $this->paginationSort;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getTermFilter()
+    {
+        return $this->termFilter;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getTimeout()
+    {
+        return $this->timeout;
+    }
+
+        /**
+     * @return Command\Indexes\Response
+     */
+    public function getResponse()
+    {
+        return $this->response;
+    }
```

Comment:
You could get around needing this method since it is in the parent Command class by just adding the response property like done here: https://github.com/basho/riak-php-client/blob/jira/am/clients-268/src/Riak/Command/DataType/Counter/Store.php#L41